tests/test_jws.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) The Diem Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+from diem import jws
+from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey, Ed25519PublicKey
+from cryptography.exceptions import InvalidSignature
+import pytest
+
+
+HEX_KEY = "ccf2700f8b2d001a8caf80ca2bfc5b7cc71df1f799e02c78bc3f07ea3af79a98"
+KEY = Ed25519PrivateKey.from_private_bytes(bytes.fromhex(HEX_KEY))
+PUBLIC_KEY = KEY.public_key()
+HEADERS = {"keyId": "hello", "alg": "EdDSA"}
+MSG = "msg"
+
+
+def test_encode_decode_message():
+    sig = jws.encode(MSG, KEY.sign)
+    headers, body = jws.decode(sig, PUBLIC_KEY.verify)
+    assert headers == {"alg": "EdDSA"}
+    assert body == MSG
+
+    assert (
+        sig
+        == b"eyJhbGciOiAiRWREU0EifQ.bXNn.xG-o01MoqLVrKuu_fCN4tHaJdpCGK2iiX8c7Tac2ENNsblrqloGQJ4H8hUF7Ikxpu3ZJCtnkJpqAsUeG2F4TDw"
+    )
```

Comment:
```
import base64 

from jwcrypto import jws, jwk
from jwcrypto.common import json_decode, json_encode

from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey

hex_key = "ccf2700f8b2d001a8caf80ca2bfc5b7cc71df1f799e02c78bc3f07ea3af79a98"
ekey = Ed25519PrivateKey.from_private_bytes(bytes.fromhex(hex_key))
key_json = {"kty":"OKP","crv":"Ed25519", "d": base64.urlsafe_b64encode( ekey._raw_private_bytes() ).rstrip(b"=").decode(), "x": base64.urlsafe_b64encode(ekey.public_key()._raw_public_bytes() ).rstrip(b"=").decode() }
key = jwk.JWK.from_json(json_encode(key_json))

s = jws.JWS("msg")
s.add_signature(key, None, {'alg': 'EdDSA'}, None)
print(s.serialize(compact=True))
```
prints `eyJhbGciOiJFZERTQSJ9.bXNn.xtWbB8A-Jp-UYspmnYYrVGgfGRzV9TCTZ5j5z7Z_y-FtsI116Jkp81_n7OkkPInOk4P9Df2X11paOXaSs_0QCQ`

Reply:
They are same now. 
The difference is protected header json, one uses: `{"alg": "EdDSA"}`, which is default python json dumps result, `jwcrypto` uses `{"alg":"EdDSA"}`, removed the space after `:`